rclcpp/include/rclcpp/executor.hpp

```diff
@@ -76,9 +77,97 @@ class Executor
   virtual void
   spin() = 0;
 
+  /// Add a callback group to an executor.
+  /**
+   * An executor can have zero or more callback groups which provide work during `spin` functions.
+   * When an executor attempts to add a callback group, the executor checks to see if it is already
+   * associated with another executor.
+   * If it is, than an exception is thrown.
```

Comment:
Please be specific about what exception type is thrown and please use `\throw` to document it.